src/python/gridmet/netCDF_components_consolidator.py

```diff
@@ -171,21 +175,23 @@ def add_component(self, filename: str, var: str = None, abs_var: str = None):
             
             logging.debug(str(self.components_list))
             self.components_list.append(var)
-            logging.debug("Shape: " + str(new_dataset[var].shape))
+            logging.debug("Shape: " + str(component_pct_dataset[var].shape))
 
         
         # Check if the grid of the component file is compatible with the grid of the existing Dataset
-        if new_dataset[var].shape != self.dataset[self.main_var].shape:
+        if component_pct_dataset[var].shape != self.dataset[self.main_var].shape:
```

Comment:
are you checking other "gis metadata" attributes other than shape? CRS and navalues should also be compatible